test/junit/scala/tools/nsc/backend/jvm/opt/BoxUnboxTest.scala

```diff
@@ -0,0 +1,325 @@
+package scala.tools.nsc
+package backend.jvm
+package opt
+
+import org.junit.Assert._
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+import scala.collection.JavaConverters._
+import scala.tools.asm.Opcodes._
+import scala.tools.asm.tree.ClassNode
+import scala.tools.nsc.backend.jvm.AsmUtils._
+import scala.tools.partest.ASMConverters._
+import scala.tools.testing.BytecodeTesting
+import scala.tools.testing.BytecodeTesting._
+
+/**
+  * Tests for boxing/unboxing optimizations.
+  */
+@RunWith(classOf[JUnit4])
+class BoxUnboxTest extends BytecodeTesting {
```

Comment:
If it's not too annoying, could you prepend a simple commit that creates this new test file without other changes? This way, the actual changes are easier to spot.